packages/react-instantsearch-dom/src/components/VoiceSearch.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { translatable, Translate } from 'react-instantsearch-core';
+import { createClassNames } from '../core/utils';
+const cx = createClassNames('VoiceSearch');
+
+type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+type ToggleListening = (toggleListeningParams: {
+  searchAsYouSpeak: boolean;
+}) => void;
+
+type InnerComponentProps = {
+  status: string;
+  errorCode?: string;
+  isListening: boolean;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  isBrowserSupported: boolean;
+};
+
+type VoiceSearchProps = {
+  isBrowserSupported: boolean;
+  isListening: boolean;
+  toggleListening: ToggleListening;
+  voiceListeningState: VoiceListeningState;
+  searchAsYouSpeak?: boolean;
+
+  translate: Translate;
+  buttonComponent?: React.FC<InnerComponentProps>;
+  statusComponent?: React.FC<InnerComponentProps>;
+};
+
+const DefaultButton: React.FC<InnerComponentProps> = ({
+  status,
+  errorCode,
+  isListening,
+}) => {
+  if (status === 'error' && errorCode === 'not-allowed') {
+    return (
+      <svg
```

Comment:
We should be able to not repeat the outer `svg` element, only the inner elements differ.